bundles/org.openhab.binding.solarwatt/src/main/java/org/openhab/binding/solarwatt/internal/handler/EnergyManagerConnector.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.solarwatt.internal.handler;
+
+import java.net.URI;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.solarwatt.internal.SolarwattConfiguration;
+import org.openhab.binding.solarwatt.internal.domain.EnergyManagerCollection;
+import org.openhab.binding.solarwatt.internal.domain.dto.EnergyManagerDTO;
+import org.openhab.binding.solarwatt.internal.exception.SolarwattConnectionException;
+import org.openhab.binding.solarwatt.internal.factory.EnergyManagerDevicesFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * Class to talk to the energy anager via HTTP and return the concrete device instances.
+ *
+ * @author Sven Carstens - Initial contribution
+ */
+@NonNullByDefault
+public class EnergyManagerConnector {
+    private static final String PROTOCOL = "http://";
+    private static final String WIZARD_DEVICES_URL = "/rest/kiwigrid/wizard/devices";
+    private static final long CONNECT_TIMEOUT_SECONDS = 30;
+
+    private final Logger logger = LoggerFactory.getLogger(EnergyManagerConnector.class);
+    private final Gson gson = new GsonBuilder().create();
+    private final HttpClient httpClient;
+    private @Nullable URI energyManagerURI;
+
+    public EnergyManagerConnector(final HttpClient httpClient) {
+        this.httpClient = httpClient;
+    }
+
+    /**
+     * Pass in the configuration to know which host to talk to.
+     *
+     * @param configuration containing the hostname.
+     */
+    public void setConfiguration(final @Nullable SolarwattConfiguration configuration) {
+        if (configuration != null) {
+            String hostname = configuration.hostname;
+
+            if (!"".equals(hostname)) {
```

Comment:
```suggestion
            if (!hostname.isEmpty()) {
```
More efficient as you dont need to first convert to a string.

Reply:
Changed.